src/bin/pgcopydb/pgsql.c

```diff
@@ -174,11 +174,9 @@ pgsql_init(PGSQL *pgsql, char *url, ConnectionType connectionType)
 
 	/* set our default retry policy for interactive commands */
 	(void) pgsql_set_interactive_retry_policy(&(pgsql->retryPolicy));
-
 	if (validate_connection_string(url))
 	{
-		/* size of url has already been validated. */
-		strlcpy(pgsql->connectionString, url, MAXCONNINFO);
+		pgsql->connectionString = url;
```

Comment:
I think this requires editing the comment of the function to say that the given `char *url` is expected to point to a string that is kept around for at least as long as the PGSQL instance.